lib/auth/auth.go

```diff
@@ -4378,6 +4378,47 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 	return req, nil
 }
 
+// getAccessRequestPromotions will return potential access list promotions for an access request.
+func (a *Server) getAccessRequestPromotions(ctx context.Context, req types.AccessRequest) (types.AccessRequest, *types.AccessRequestAllowedPromotions) {
```

Comment:
Do we really need to return request copy from this function? I don't think we actually modify the request, so calling function can also just do `req.Copy()` itself. It will require one more `Copy()` call but unless it's really expensive, I think code will look better.

Reply:
AFAICT no, but I didn't want to change things in this regard without knowing for certain that this request was guaranteed not to be changed. Given that this copy is being made, I'm guessing it's possible this could modify the original request?